pkg/controller/data_upload_controller.go

```diff
@@ -593,18 +709,18 @@ func (r *DataUploadReconciler) onPrepareTimeout(ctx context.Context, du *velerov
 	r.metrics.RegisterDataUploadFailure(r.nodeName)
 }
 
-func (r *DataUploadReconciler) exclusiveUpdateDataUpload(ctx context.Context, du *velerov2alpha1api.DataUpload,
-	updateFunc func(*velerov2alpha1api.DataUpload)) (bool, error) {
-	updated := du.DeepCopy()
-	updateFunc(updated)
+func ExclusiveUpdateDataUpload(ctx context.Context, client client.Client, du *velerov2alpha1api.DataUpload,
+	updateFunc func(*velerov2alpha1api.DataUpload)) (*velerov2alpha1api.DataUpload, bool, error) {
+	original := du.DeepCopy()
+	updateFunc(du)
```

Comment:
`ExclusiveUpdateDataUpload` is a fundermental function, we don't expect all the callers expect a updated `du`, so it is risky to update the `du` directly.

Actually, I suggest we leave `ExclusiveUpdateDataUpload` as is, and change the function of `acceptDataUpload`, the latter is more pure and it is safe to change it.

Reply:
I've changed back to exclusiveUpdateDataDownload.

But for this function, in the update action, we needn't copy it before updating it for general usage, leading to many related UT tests failing with[ conflict errors](https://stackoverflow.com/questions/51297136/kubectl-error-the-object-has-been-modified-please-apply-your-changes-to-the-la). 

So I removed the DeepCopy function for the UT, I know your original purpose would be not willing to update the `du` directly, but for our current usage scenario there is no effect as we always get the latest CR from the API server instead of memory in each Reconcile